go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -171,7 +171,7 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 			// Call UpdateTabletFields instead of ChangeType so that we can check the type
 			// before changing it and avoid unnecessary topo updates
 			var err error
-			oldMasterTablet, err = topotools.ChangeType(ctx, agent.TopoServer, oldMasterAlias, topodatapb.TabletType_REPLICA, nil)
+			oldMasterTablet, err = topotools.ChangeType(ctx, agent.TopoServer, oldMasterAlias, agent.DemoteMasterType, nil)
```

Comment:
Once all components are upgraded to 4.0.0+, this RPC doesn't get called any more. It is here for backwards compatibility during rollouts. It's best to not make any changes to it.

Reply:
Done, reverted changes in this RPC path